lkmpg.tex

```diff
@@ -219,17 +219,44 @@ \subsection{The Simplest Module}
 \begin{code}
 obj-m += hello-1.o
 
+PWD := $(CURDIR)
+
 all:
 	make -C /lib/modules/$(shell uname -r)/build M=$(PWD) modules
 
 clean:
 	make -C /lib/modules/$(shell uname -r)/build M=$(PWD) clean
 \end{code}
 
-And finally just:
+And finally just run \verb|make| directly.
 \begin{codebash}
 make
 \end{codebash}
+If there is no \verb|PWD := $(CURDIR)| statement in Makefile, note that using \verb|sudo make| might not compile. Because the \verb|sudo| will fork the process. According to the Process Model section in \href{https://man7.org/linux/man-pages/man8/sudo.8.html}{sudo manual}, some of the environment variables won't be inherited. Here is the \verb|PWD| variable which will be passed to the kernel Makefile. For example, here is the simple Makefile:
```

Comment:
Add the empty lines between those explanations and code sections.

Reply:
Separate the sentences into different lines.
For example:
```latex
235 If there is no \verb|PWD := $(CURDIR)| statement in Makefile, note that using \verb|sudo make| might not compile.
236 Because the \verb|sudo| ...
```

Reply:
```latex
... Because the \verb|sudo| will fork the process. ...
```
Change this incorrect description to the security policy `sudoers` that I mentioned previously.

Reply:
Thanks for your advice. It looks like I need to clone the project to the local then modify, before I used to change it on the web.